src/server/main_service.cc

```diff
@@ -2040,7 +2042,9 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
         "Dragonfly does not allow execution of a server-side Lua in Multi transaction");
 
   bool scheduled = false;
-  if (*multi_mode != Transaction::NOT_DETERMINED) {
+  if (*multi_mode == Transaction::NOT_DETERMINED) {
+    cntx->transaction->InitTxTime();
```

Comment:
how about call this in transaction constructor and if we will do scheduling this will overide the value

Reply:
That's a fantastic idea!